### DISCUSSION TITLE
DC Grid Feed IN and DVCC

### DISCUSSION THREAD

**Discussion by peregrines2** [Louisvdw/dbus-serialbattery#493] [General]
Hi!

I just found out why I am always running into BMS Emergency-Turn offs and my MPPTs dont stop charging my batteries even after reaching my SOC High Limits.
Enabling DC Grid Feed in dissables DVCC settings for cell voltages.

How do you guys handle that? I need to DC-Feed to the grid when my batteries are "full" because I dont want to loose a big amount of energy with my two MPPT 450/100. 

Is there any "workaround" to export excess energy to the grid and stop charing my batteries with DC Feed in turned ON?
I am using Homeassistant and have Modbus and MQTT Setup - maybe there is some way to program something as a automation?

**Comment by redmoon2711** [comment]
Hi, 

you can try to set CVL (Charge Voltage Limit) to Float_Voltage (3.35V per cell) when the battery is full.

This will stop the charging process and the current flows from MPPT into grid.

**Reply by peregrines2** [reply]
Thank You for your help!
The only question now is, if Float Voltage can be set by MQTT or Modbus?
I cant find anything about it. Because if not, I need to manually change that setting everyday, as soon, as my battery gets to 100%. And then change it back also.

**Reply by redmoon2711** [reply]
Why is your BMS turn off?
I think that was the problem.

When you charge with CVL = 55.2V (3.45V per cell lifepo4) and your battery is balance at the topvoltage, there is no reason to turn off for the BMS.
But if one cell over 3.65V, it must protect your battery and turn off.

The Multiplus feed the energy into the grid, when the CVL <= battery-voltage.

You can set the CVL (limit charge voltage) in DVCC       or you set:

#Charge voltage control management enable (True/False).
CVCM_ENABLE = True

in utils.py and reboot. (v14.3)

The charge voltage contol management will charge your battery with 3.45V per cell (CVL). 
If the battery-voltage reached CVL,
- the multiplus feed into grid (DC feed in is enable)
- the mppt lower the current (DC feed in is disable)
- absorption starts (MAX_VOLTAGE_TIME_SEC = 15*60)

After absorption the CVL switch to 3.35V per cell (FLOAT_CELL_VOLTAGE).
The Battery is fully charged.

**Reply by peregrines2** [reply]
Thank You for your answer @redmoon2711 (I just rediscovered it).

I set everything according to your suggestions (CVCM_ENABLE = True, Cell Max is 3,45V and float is 3.35V)
As soon as the batteries reach CVL ( 55,2V) both MPPT limited their production to zero.
My AC inverters keep producing and the Multiplus export their exccess energy to grid ... just my MPPT stop completely :( 
Do you have any other settings I could check?
What Cell max and float Values should I set for my MPPT in Victron Connect and what values for my Multiplus in Ve.Config?
I would be extremely happy to sort this out and be able to export to grid again!!
Thank You so much @redmoon2711 ! 

<img width="1183" alt="Bildschirmfoto 2024-01-29 um 13 35 30" src="https://github.com/Louisvdw/dbus-serialbattery/assets/9152610/2f42079b-c0f7-40b0-8a07-21fd36c488c3">
<img width="492" alt="Bildschirmfoto 2024-01-29 um 13 37 58" src="https://github.com/Louisvdw/dbus-serialbattery/assets/9152610/a03488fa-47e5-4d90-ae5a-02f6f2619e58">

**Reply by snow78** [reply]
which version of driver are you using. i use 1.0.20240102beta. have as similar setup as yours, and it works.
you have step mode active. you should use LINEAR_LIMITATION_ENABLE = True in config.ini
this will gracefully ramp down from full charge voltage to float
you can see in your picture that the CVL is still 55.2 which is in a 16 cell LIFEPO4 to much for a float. your cells will be likely overcharged
LINEAR_LIMITATION_ENABLE = True has also advantage that it prevents single cell overvoltage when you have inbalanced packs. it will reduce the overall charge voltage
but why its not feeding to grid i dont know. setting is turned on? you do not use aggregatebatteries by any chance?

**Reply by peregrines2** [reply]
Thank You snow78!

I updated to the newest beta and set CVCM_ENABLE = True
Every since it is working.
I also set LINEAR_LIMITATION_ENABLE = True for smoother ramping.

**Reply by Ray356** [reply]
@peregrines2, 

how do you handle that with the mppt's?
If i turn on "DC Grid Feed in" my mppts don't care about he maximum charge current which is coming from the driver.

**Reply by snow78** [reply]
@Ray356 this is normal that is general behaves as designed in the VENUS. this is not an issue of the serial driver. i have the same. when DC Grid feed in is active it will "ignore" the max currect from the BMS signaled. the logic i understand behind it ... to have Grid feed in possible the MPPTs are basically not limited... but then the Batteries will surge as much current as they can get and only what the batteries do not take is feed by the e.g Multiplus to grid. limiting of charge current is bassically only possible via charge voltage limitation which kicks in with serial driver when e.g. cell overvoltage is detected. the way i solved it is by having a large enough battery (i have 3 banks with 16s *280ah) so that mppts never can exceed the max charge current.
